service/dap/server.go

```diff
@@ -357,19 +357,21 @@ func (s *Server) serveDAPCodec() {
 	s.reader = bufio.NewReader(s.conn)
 	for {
 		request, err := dap.ReadProtocolMessage(s.reader)
-		// TODO(polina): Differentiate between errors and handle them
-		// gracefully. For example,
+		// Handle errors gracefully by responding with an ErrorResponse.
 		// -- "Request command 'foo' is not supported" means we
 		// potentially got some new DAP request that we do not yet have
 		// decoding support for, so we can respond with an ErrorResponse.
-		// TODO(polina): to support this add Seq to
-		// dap.DecodeProtocolMessageFieldError.
 		if err != nil {
 			select {
 			case <-s.stopTriggered:
 			default:
 				if err != io.EOF {
-					s.log.Error("DAP error: ", err)
+					// Send an error response to the users if we were unable to process the message.
+					var seq int
+					if decodeErr, ok := err.(*dap.DecodeProtocolMessageFieldError); ok {
+						seq = decodeErr.Seq
+					}
+					s.sendDapErrorResponse(seq, err.Error())
```

Comment:
Actually, this is exactly the kind of thing that sendInternalErrorResponse was added for. It is used for similar things already in the code (except for the one instance in onLaunchRequest, where it is used incorrectly, but that one is going away as I am refactoring this code right now).

Reply:
Done.